Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -136,20 +136,44 @@ private func edits(from original: DocumentSnapshot, to edited: String) -> [TextE
 
 extension SwiftLanguageService {
   package func documentFormatting(_ req: DocumentFormattingRequest) async throws -> [TextEdit]? {
-    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    return try await format(
+      textDocument: req.textDocument,
+      options: req.options
+    )
+  }
+
+  package func documentRangeFormatting(_ req: DocumentRangeFormattingRequest) async throws -> [TextEdit]? {
+    return try await format(
+      textDocument: req.textDocument,
+      options: req.options,
+      extraArgs: [
+        "--offsets",
+        "\(req.range.lowerBound.line):\(req.range.upperBound.line)",
+      ]
```

Comment:
I would prefer to keep the assembly of the swift-format command line invocation at one position and thus pass the range to format to `format` instead of an extra command line option.